components/scanners/trufflehog/internal/transformer/transformer.go

```diff
@@ -298,12 +313,19 @@ func (t *trufflehogTransformer) mapDataSource(ctx context.Context, location Truf
 	if location.SourceMetadata.Data.Filesystem.File == "" {
 		return "", errors.Errorf("unsupported trufflehog findings with empty file entry detected, finding %#v", location)
 	}
+
+	fullPath := location.SourceMetadata.Data.Filesystem.File
+	if !strings.HasPrefix(fullPath, t.stripFilePathPrefix) {
+		return "", errors.Errorf("absolute path %q does not have expected prefix %q", fullPath, t.stripFilePathPrefix)
+	}
+	relativePath := strings.TrimPrefix(fullPath, t.stripFilePathPrefix)
+
 	targetMetadata := component.TargetMetadataFromCtx(ctx)
 	dataSource := ocsffindinginfo.DataSource{
 		TargetType: t.targetType,
 		Uri: &ocsffindinginfo.DataSource_URI{
 			UriSchema: ocsffindinginfo.DataSource_URI_SCHEMA_FILE,
-			Path:      location.SourceMetadata.Data.Filesystem.File,
```

Comment:
does this need to be `Path: utils.Ptr(fmt.Sprintf("file://%s", relativePath)),` like the affectedCode block?? I don't have context why we didn't add the file:// bit from the beginning @northdpole 

thank you 🙏🏻